app/src/components/market/common/history_section/index.tsx

```diff
@@ -0,0 +1,173 @@
+        Object.values(queryResult.data).forEach(value => {
+          if (value && value.outcomeTokenAmounts) {
+            const block = blocks[index]
+            const holdings = value.outcomeTokenAmounts
+            result.push({ block, holdings })
+          }
+        })
+      })
+
+    return result
+  }
+  return null
+}
+
+type Props = {
+  answerFinalizedTimestamp: Maybe<BigNumber>
+  marketMakerAddress: string
+  hidden: boolean
+  outcomes: string[]
+  oracle: Maybe<string>
+  currency: string
+  fee: BigNumber
+  decimals: number
+  scalarHigh: Maybe<BigNumber>
+  scalarLow: Maybe<BigNumber>
+  unit: string
+}
+
+const blocksPerAllTimePeriod = 10000
+const blocksPerDay = 5760
+const blocksPerHour = Math.floor(blocksPerDay / 24)
+const blocksPerMinute = Math.floor(blocksPerHour / 60)
+
+const calcOffsetByDate = (nowOrClosedTs: number) => {
+  const now = moment()
+  const offsetInMinutes = moment(nowOrClosedTs)
+    .startOf('day')
+    .diff(now, 'minute')
+
+  return -offsetInMinutes * blocksPerMinute
+}
+
+export const HistorySelectContainer: React.FC<Props> = ({
+  answerFinalizedTimestamp,
+  currency,
+  decimals,
+  fee,
+  hidden,
+  marketMakerAddress,
+  oracle,
+  outcomes,
+  scalarHigh,
+  scalarLow,
+  unit,
+}) => {
+  const context = useWeb3Context()
+  const { library } = context
+  const [latestBlockNumber, setLatestBlockNumber] = useState<Maybe<number>>(null)
+
+  const [blocks, setBlocks] = useState<Maybe<Block[]>>(null)
+  const holdingsSeries = useHoldingsHistory(marketMakerAddress, blocks)
+  const [period, setPeriod] = useState<Period>('1M')
+
+  const blocksOffset = useMemo(
+    () => (answerFinalizedTimestamp ? calcOffsetByDate(answerFinalizedTimestamp.toNumber() * 1000) : 0),
+    [answerFinalizedTimestamp],
+  )
+
+  const blocksSinceInception = latestBlockNumber ? latestBlockNumber - EARLIEST_MAINNET_BLOCK_TO_CHECK : 0
+  const allDataPoints = Math.floor(blocksSinceInception / blocksPerAllTimePeriod)
+
+  const mapPeriod: { [period in Period]: { totalDataPoints: number; blocksPerPeriod: number } } = {
+    All: { totalDataPoints: allDataPoints, blocksPerPeriod: blocksPerAllTimePeriod },
+    '1Y': { totalDataPoints: 365, blocksPerPeriod: blocksPerDay },
+    '1M': { totalDataPoints: 30, blocksPerPeriod: blocksPerDay },
+    '1W': { totalDataPoints: 7, blocksPerPeriod: blocksPerDay },
+    '1D': { totalDataPoints: 24, blocksPerPeriod: blocksPerHour },
+    '1H': { totalDataPoints: 60, blocksPerPeriod: blocksPerMinute },
+  }
+
+  useEffect(() => {
+    library.getBlockNumber().then((latest: number) => setLatestBlockNumber(latest - blocksOffset))
+  }, [blocksOffset, library])
+
+  useEffect(() => {
+    const getBlocks = async (latestBlockNumber: number) => {
+      const { blocksPerPeriod, totalDataPoints } = mapPeriod[period]
+
+      if (latestBlockNumber) {
+        const blockNumbers = range(totalDataPoints).map(multiplier => latestBlockNumber - multiplier * blocksPerPeriod)
```

Comment:
This line throws an error when switching to "All" in the filter picker and the app crashes, because the totalDataPoints is negative so the range function throws an error.
I am guessing that it has to do with the datapoints returned by "All" which is negative if the market hasn't existed for a long time.

Steps to reproduce
Switch to Rinkeby tab
Go to market /#/0xf9cefa4ade74834882589f5309664c35ecdeff5f
History->Graph.
Click a few date range filters and finally click the "All" filter.
![image](https://user-images.githubusercontent.com/4677914/104459814-dab07f00-55df-11eb-9d5c-9a29accaecce.png)

Reply:
I made issue for that that is on master as well. Here is the issue https://github.com/protofire/omen-exchange/issues/1536

Reply:
So I guess you'll either need to handle it on your side or wait for a fix to be made on the other side before deploying.

Reply:
Yeah but that is not the error caused by this pr and this pr has been expanded enough already.And it won't make any difference since it already has the same error on master...